socrates/lib/notifications/index.js

```diff
@@ -23,71 +26,143 @@ function renderingOptions(member) {
-    if (err || !receivers) { return logger.error(err); }
-    var file = path.join(__dirname, 'pug/' + organizersFilename + '.pug');
-    notifications._sendMail(receivers, organizersSubject, pug.renderFile(file, options));
+function notifyConcernedParties(params, self) {
+  var options = renderingOptions(params.member);
+  options.bookingdetails = params.bookingdetails;
+  var filename = path.join(__dirname, 'pug/' + params.participantFilename + '.pug');
+  sendMail({
+    fromName: 'SoCraTes Notifications',
+    fromAddress: self.registrationListEmailAddress(),
+    receivers: [params.member.email()],
+    subject: params.participantSubject,
+    html: pug.renderFile(filename, options)
+  });
+
+  var receivers = self.registrationListEmailAddress();
+  if (!receivers) { return logger.error('no receivers for sending mail'); }
+  filename = path.join(__dirname, 'pug/' + params.organizersFilename + '.pug');
+  sendMail({
+    fromName: 'SoCraTes Notifications',
+    fromAddress: conf.get('sender-address'),
+    receivers: [receivers],
+    subject: params.organizersSubject,
+    html: pug.renderFile(filename, options)
   });
 }
+
 module.exports = {
   newSoCraTesMemberRegistered: function (member) {
-    membersService.registrationNotificationEmailAddresses(function (err, receivers) {
-      if (err || !receivers) { return logger.error(err); }
-      subscriberstore.allSubscribers(function (err1, subscribers) {
-        if (err1 || !subscribers) { return logger.error(err1); }
-        var options = renderingOptions(member);
-        options.count = subscribers.length;
-        var filename = path.join(__dirname, 'pug/newmembertemplate.pug');
-        notifications._sendMail(receivers, 'Neuer Interessent', pug.renderFile(filename, options));
+    var receivers = this.registrationListEmailAddress();
+    if (!receivers) { return logger.error('no receivers for sending mail'); }
+    subscriberstore.allSubscribers(function (err1, subscribers) {
+      if (err1 || !subscribers) { return logger.error(err1); }
+      var options = renderingOptions(member);
+      options.count = subscribers.length;
+      var filename = path.join(__dirname, 'pug/newmembertemplate.pug');
+      sendMail({
+        fromName: 'SoCraTes Notifications',
+        fromAddress: conf.get('sender-address'),
+        receivers: [receivers],
+        subject: 'Neuer Interessent',
+        html: pug.renderFile(filename, options)
       });
     });
   },
 
   newParticipant: function (memberID, bookingdetails) {
+    const self = this;
     memberstore.getMemberForId(memberID, function (err, member) {
       if (err || !member) { return logger.error(err); }
-      notifyConcernedParties(member, bookingdetails, 'registrationConfirmation', 'SoCraTes Registration Confirmation',
-        'superuserRegistrationNotification', 'New SoCraTes Registration');
+      var params = {
+        member: member,
+        bookingdetails: bookingdetails,
```

Comment:
nein, das geht nur bei denen, wo key === value. Ist in diesem Fall also etwas ungleichmäßig. Bin auch nicht so sicher, was hier schöner ist.